### PR TITLE
snprintf should be defined in time_stamp.c only for MSVC 2013

### PR DESCRIPTION
**Pull request by fbudin69500**

### CODE REVIEW

**Comment by fbudin69500** [comment]
@thewtex My previous PR actually broke things with MSVC 2015. Moving the `#define` after including `stdio.h` solves the problem, but to avoid any other issue, I added a check of the version of Visual Studio. Potentially it could even be limited to VIsual Studio 2012 instead of VS2012 and below. Let me know what you think.

File: libcommon/time_stamp.c [hunk 1]
```diff
@@ -62,17 +62,20 @@
...
-#endif
-#endif
-
 #ifdef HAVE_STRING_H
 #include <string.h>
 #endif
 #include <stdio.h>
 
+// MS Visual Studio 12 and below.
+// Declared after include of `stdio.h` in case it is already devined in that
```
Comment (thewtex) [inline libcommon/time_stamp.c]: defined

**Comment by thewtex** [comment]
Build errors identified by Travis.

**Comment by fbudin69500** [comment]
@thewtex The error reported by travis does not seem related to my commit. It has to do with `Cound not find NetCDF`, on MacOS. Any idea why?

**Comment by thewtex** [comment]
@fbudin69500 I recall a different error, but I do not see it now. Marching forward...

**Pull request merged**